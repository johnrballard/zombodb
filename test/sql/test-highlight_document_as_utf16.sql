create table ft
(
    ft   zdb.fulltext,
    ftws zdb.fulltext_with_shingles
);
insert into ft (ft, ftws)
values ('The map described sailing 23° North by 17° East to reach the island.
The name of the island is ⅋☺︎＠¡¿ To retrieve the treasure, find the largest
tree behind BestBuy® and click the treasure chest icon.',
        'The map described sailing 23° North by 17° East to reach the island.
The name of the island is ⅋☺︎＠¡¿ To retrieve the treasure, find the largest
tree behind BestBuy® and click the treasure chest icon.');

create index idxft on ft using zombodb ((ft.*));

select *
from zdb.highlight_document('ft', (select to_json(ft.*) from ft),
    'ft:treasure or ftws:treasure')
order by start_offset, field_name;

drop table ft;