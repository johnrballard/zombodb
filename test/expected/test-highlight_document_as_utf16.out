create table ft
(
    ft   zdb.fulltext,
    ftws zdb.fulltext_with_shingles
);
insert into ft (ft, ftws)
values ('The map described sailing 23° North by 17° East to reach the island.
The name of the island is ⅋☺︎＠¡¿ To retrieve the treasure, find the largest
tree behind BestBuy® and click the treasure chest icon.',
        'The map described sailing 23° North by 17° East to reach the island.
The name of the island is ⅋☺︎＠¡¿ To retrieve the treasure, find the largest
tree behind BestBuy® and click the treasure chest icon.');
create index idxft on ft using zombodb ((ft.*));
select *
from zdb.highlight_document('ft', (select to_json(ft.*) from ft),
    'ft:treasure or ftws:treasure')
order by start_offset, field_name;
 field_name | array_index |   term   |    type    | position | start_offset | end_offset |  query_clause   
------------+-------------+----------+------------+----------+--------------+------------+-----------------
 ft         |           0 | treasure | <ALPHANUM> |       23 |          118 |        126 | ft:"treasure"
 ftws       |           0 | treasure | <ALPHANUM> |       23 |          118 |        126 | ftws:"treasure"
 ft         |           0 | treasure | <ALPHANUM> |       33 |          180 |        188 | ft:"treasure"
 ftws       |           0 | treasure | <ALPHANUM> |       33 |          180 |        188 | ftws:"treasure"
(4 rows)

drop table ft;
